### PR TITLE
v3.1.1 add subscription and gopay tokenization API feature

### PR DESCRIPTION
**Pull request by Xaxxis**
- add subscription and gopay tokenization API
- fix error handling from midtrans satus_code
- fix logging interceptor not print custom header 
- add sample tokenization controller for SDK Client

### CODE REVIEW

File: library/src/main/java/com/midtrans/httpclient/APIHttpClient.java [hunk 1]
```diff
@@ -116,12 +121,12 @@ private static OkHttpClient buildHttpClient(Config config) {
         // setup default http client
         if (config.getProxyConfig() == null) {
             return httpClient.newBuilder()
-                    .addInterceptor(loggingInterceptor(config))
                     .connectTimeout(config.getConnectionTimeout(), config.getHttpClientTimeUnit())
                     .readTimeout(config.getReadTimeout(), config.getHttpClientTimeUnit())
                     .writeTimeout(config.getWriteTimeout(), config.getHttpClientTimeUnit())
                     .connectionPool(connectionPool)
                     .addInterceptor(headersInterceptor)
+                    .addInterceptor(loggingInterceptor(config))
```
Comment (rizdaprasetya) [inline library/src/main/java/com/midtrans/httpclient/APIHttpClient.java:129]: There is 2 `addInterceptor` here, will the bottom overwrite the top one? or both will be added?
If it get overwritten, then it seems like a potential bug

**Review by rizdaprasetya — APPROVED** [review]
Nice. Looks good to me, no major issue. Just that 1 small question. Then you can proceed to merge & release as new version. Nice work @Xaxxis ! 👍

File: library/src/main/java/com/midtrans/httpclient/APIHttpClient.java (hunk 1 above)
Comment (Xaxxis) [reply library/src/main/java/com/midtrans/httpclient/APIHttpClient.java:129]: Yes, that's expected, both will be added @rizdaprasetya

Comment (rizdaprasetya) [reply library/src/main/java/com/midtrans/httpclient/APIHttpClient.java:129]: cool, that will be fine 👌

**Pull request merged**